### PR TITLE
Metrics : Introduce fallback mechanism when no watermark was coalesced

### PR DESCRIPTION
**Pull request by Fly-Style**
Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by barisbasaran — APPROVED** [review]
LGTM!

File: hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java [hunk 1]
```diff
@@ -720,12 +720,21 @@ public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollection
...
+                mContext.collect(keyedDesc, LAST_FORWARDED_WM, ProbeLevel.INFO, ProbeUnit.MS, outbox.lastForwardedWm(key));
+                long lastForwardedWmLatency = lastForwardedWmLatency(key);
+                mContext.collect(keyedDesc, LAST_FORWARDED_WM_LATENCY, ProbeLevel.INFO, ProbeUnit.MS, lastForwardedWmLatency);
+            }
+        } else {
+            MetricDescriptor keyedDesc = descriptor.copy().withDiscriminator("key", "0");
+            mContext.collect(keyedDesc, TOP_OBSERVED_WM, ProbeLevel.INFO, ProbeUnit.MS, Long.MIN_VALUE);
+            mContext.collect(keyedDesc, COALESCED_WM, ProbeLevel.INFO, ProbeUnit.MS, Long.MIN_VALUE);
+            mContext.collect(keyedDesc, LAST_FORWARDED_WM, ProbeLevel.INFO, ProbeUnit.MS, Long.MIN_VALUE);
+            mContext.collect(keyedDesc, LAST_FORWARDED_WM_LATENCY, ProbeLevel.INFO, ProbeUnit.MS, 0L);
```
Comment (ramizdundar) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java:737]: Why do we need this values? Can't we just not collect anything if keys are empty?

Comment (ramizdundar) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java:737]: To rephrase I understand what this PR does and it's correct, but I don't understand why.

Comment (Fly-Style) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java:737]: @barisbasaran @zhmaeff knows more, it's MC side specifics.

Comment (barisbasaran) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java:737]: what do you think @zhmaeff ?

Comment (zhmaeff) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java:737]: Without these changes MC couldn't find some metrics that were present before. So it is just for preserving previous behavior from the MC perspective.

**Review by viliam-durina — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java (hunk 1 above)
Comment (ramizdundar) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java:737]: Thanks.

**Review by ramizdundar — APPROVED** [review]

**Comment by barisbasaran** [comment]
@Fly-Style shall we merge this one? It's kind of blocking our builds

**Pull request merged**